src/datasources/cache/cache.service.interface.ts

```diff
@@ -11,5 +11,7 @@ export interface ICacheService {
 
   get(cacheDir: CacheDir): Promise<string | undefined>;
 
-  delete(cacheDir: CacheDir): Promise<number>;
+  deleteByKey(key: string): Promise<number>;
+
+  deleteByKeyPattern(pattern: string): Promise<void>;
```

Comment:
I changed `delete` function name to `deleteByKey` to be more explicit on the expected argument and the purpose of the function. Previously this `delete` function was accepting a `CacheDir` instead of a `string`. Here I think there is no perfect solution:
- If a `CacheDir` is accepted, then the client is forced to send a `CacheDir` which will have `key` and `field`. The `delete` function will delete by `key` and ignore the `field`, so the purpose of the `delete` function could be obscure to the client using the class API (will be forced to set an empty `field`). Anything could be placed in the `field` since it will get ignored by the `delete` function which is very weird from the client's point of view.
- If a `string` is accepted, the purpose is clearer, but it makes the class API a bit odd (some functions accept `CacheDir` and others accept `string`), and the internal structure of `CacheDir` is exposed.
I sightly prefer the second approach based on its clarity, but of course, this is an open matter.

Reply:
Let's do as you suggested (2nd approach). Any potential improvements would probably require some work on `CacheDir` itself.